packaging/mgmtworker/files/opt/mgmtworker/scripts/fetch-logs

```diff
@@ -0,0 +1,117 @@
+#! /opt/mgmtworker/env/bin/python3
+import argparse
+import os
+import sys
+from tempfile import mkdtemp, NamedTemporaryFile
+
+import requests
+import shutil
+
+from cloudify.zip_utils import make_zip64_archive
+
+BASE_URL = 'https://{server}:8009/cfylogs/'
+
+
+def get_logs_dir(server, path, data_dir, log_handle, session):
+    save_path = os.path.join(data_dir, server, path)
+    url = BASE_URL.format(server=server) + path
+
+    contents = session.get(url)
+
+    os.makedirs(save_path, exist_ok=True)
+
+    if contents.status_code != 200:
+        log_handle.write(
+            f'Failed to retrieve dir {path}: '
+            f'{contents.status_code}- {contents.reason}\n'
+        )
+        return
+
+    files = [entry['name'] for entry in contents.json()
+             if entry['type'] == 'file']
+
+    dirs = [entry['name'] for entry in contents.json()
+            if entry['type'] == 'directory']
+
+    for file_name in sorted(files):
+        downloaded_file = session.get(url + file_name)
+        file_path = path + file_name
+
+        if downloaded_file.status_code == 200:
+            log_handle.write(f'Retrieved file {file_path}\n')
+            with open(os.path.join(save_path, file_name), 'w',
+                      encoding='utf8') as file_handle:
+                file_handle.write(downloaded_file.text)
+        else:
+            log_handle.write(
+                f'Failed to retrieve file {file_path}: '
+                f'{downloaded_file.status_code}- {downloaded_file.reason}\n'
+            )
+
+    for dir_name in sorted(dirs):
+        get_logs_dir(server, path + dir_name + '/',
+                     data_dir, log_handle, session)
+
+
+def get_logs(servers, ca_cert, credentials):
+    try:
+        temp_path = mkdtemp()
```

Comment:
but actually, just use `with tempfile.TemporaryDirectory() as temp_path`

Reply:
I was trying to remember why I chose this route, but yeah- I should do that. Will update.